### PR TITLE
Smarty Include Handler

### PR DESCRIPTION
**Pull request by ssigwart**

### CODE REVIEW

File: src/chrome/komodo/content/hyperlinks/filehandler.js [hunk 1]
```diff
@@ -210,6 +210,35 @@
...
+                filepath = smartyTemplatePathMatch[1] + filepath;
+        }
+
+        // Open file
+        filename_jump_handler(filepath);
+    }
+    ko.hyperlinks.handlers.smartyIncludeHandler =
+        new ko.hyperlinks.RegexHandler(
+            "Smarty include handler",
+            new RegExp("{include\\s+(.*\\s)?file=[\"']([^'\"}]+)[\"']", "i"),
```
Comment (cgchoffman) [inline src/chrome/komodo/content/hyperlinks/filehandler.js]: I assume you've tested this so you know it works but why is `\\s` working in this context?  I tried it with the example you provided in the Komodo regex editor and it doesn't work.  If I remove the first `\` then it works fine.  If you've tested it and it works the it's fine with me, bonus points if you tell me why though!

File: src/chrome/komodo/content/hyperlinks/filehandler.js [hunk 2]
```diff
@@ -210,6 +210,35 @@
...
+     * A Smarty include file handler - to help in opening files. Examples:
+     *    {include file="path/to/file.tpl"}
+     */
+    function smarty_jump_handler(match) {
+        var filepath = match[2];
+
+        // If not absolute path, see if there's a "/templates/" parent of current file.
+        if (filepath[0] != "/")
+        {
+            var view_filepath = ko.views.manager.currentView.koDoc.file.path;
```
[the remark this answers is not in the document]
Comment (ssigwart) [reply src/chrome/komodo/content/hyperlinks/filehandler.js]: Updated.

File: src/chrome/komodo/content/hyperlinks/filehandler.js (hunk 1 above)
Comment (ssigwart) [reply src/chrome/komodo/content/hyperlinks/filehandler.js]: Yes, it works with double slash and not with a single.  I think the reason is that we are in a Javascript string, so `\s` becomes `s` meaning regex searches for an "s."  `\\s` becomes `\s`, which regex searches for whitespace.

Comment (cgchoffman) [reply src/chrome/komodo/content/hyperlinks/filehandler.js]: @ssigwart Sorry for the delay!  Right, you jogged my memory.  You are right about it being a string in JS.  Can you wrap it in backticks and see if that fixes the issues?

Comment (cgchoffman) [reply src/chrome/komodo/content/hyperlinks/filehandler.js]: So this
```javascript
new RegExp(`{include\\s+(.*\\s)?file=[\"']([^'\"}]+)[\"']`, "i")
```

Comment (cgchoffman) [reply src/chrome/komodo/content/hyperlinks/filehandler.js]: Nvm it doesn't work.

**Pull request merged**